buildshiprun/handler.go

```diff
@@ -531,3 +566,48 @@ func getMemoryLimit() string {
 
 	return fmt.Sprintf("%s%s", unit, suffix)
 }
+
+func reportGitLabStatus(status *sdk.Status) {
+
+	payloadSecret, secretErr := sdk.ReadSecret("payload-secret")
+	if secretErr != nil {
+		log.Printf("unexpected error while reading secret: %s", secretErr)
+	}
+
+	suffix := os.Getenv("dns_suffix")
+	gatewayURL := os.Getenv("gateway_url")
+	gatewayURL = sdk.CreateServiceURL(gatewayURL, suffix)
+
+	statusBytes, marshalErr := json.Marshal(status)
+	if marshalErr != nil {
+		log.Printf("error while marshalling request: %s", marshalErr.Error())
+	}
+
+	statusReader := bytes.NewReader(statusBytes)
+	req, reqErr := http.NewRequest(http.MethodPost, gatewayURL+"function/gitlab-status", statusReader)
+	if reqErr != nil {
+		log.Printf("error while making request to gitlab-status: `%s`", reqErr.Error())
+	}
+
+	digest := hmac.Sign(statusBytes, []byte(payloadSecret))
+	req.Header.Add(sdk.CloudSignatureHeader, "sha1="+hex.EncodeToString(digest))
+
+	client := http.Client{}
+
+	res, resErr := client.Do(req)
+	if resErr != nil {
+		log.Printf("unexpected error while retrieving response: %s", resErr.Error())
+	}
+	if res.Body != nil {
+		defer res.Body.Close()
+	}
+	if res.StatusCode != http.StatusOK {
+		log.Printf("unexpected status code: %d", res.StatusCode)
+	}
+
+	_, bodyErr := ioutil.ReadAll(res.Body)
+	if bodyErr != nil {
+		log.Printf("unexpected error while reading response body: %s", bodyErr.Error())
+	}
+	status.CommitStatuses = make(map[string]sdk.CommitStatus)
```

Comment:
Required: Let's do this with `status.Clear()` instead.

Optional: Consider using a pointer reference instead of the value, so that we don't by accident update a value and miss the update.